examples/mugen/test/retrieval/test_video_clip.py

```diff
@@ -82,3 +100,62 @@ def test_forward(self, start):
         assert_expected(
             actual=out.sum(), expected=torch.as_tensor(expected_sum), rtol=0, atol=1e-3
         )
+        assert next(encoder.parameters()).requires_grad
+
+    def test_untrainable(self):
+        encoder = VideoEncoder(trainable=False)
+        assert not next(encoder.parameters()).requires_grad
+
+
+class TestProjection:
+    @pytest.fixture(autouse=True)
+    def set_seed(self):
+        set_rng_seed(1234)
+
+    @pytest.fixture
+    def utils(self, set_seed):
+        input = torch.randint(10, (2, 7)).float()
+        proj = Projection(in_dim=7, out_dim=3)
+        return proj, input
+
+    def test_forward(self, utils):
+        proj, input = utils
+        out = proj(input)
+        expected = torch.Tensor([[-1.2214, -0.0066, 1.2280], [-1.3886, 0.4626, 0.9260]])
+        assert_expected(actual=out, expected=expected, rtol=0, atol=1e-4)
+
+
+class TestVideoCLIPModel:
+    @pytest.fixture(autouse=True)
+    def set_seed(self):
+        set_rng_seed(1234)
+
+    @pytest.fixture
+    def utils(self, set_seed):
+        input_text = torch.Tensor(
+            [
+                [101, 6315, 3793, 7099, 2005, 5604, 19204, 17629, 102],
+                [101, 2117, 7820, 3793, 102, 0, 0, 0, 0],
+            ]
+        ).to(dtype=int)
+        input_video = torch.randint(10, [2, 3, 32, 32, 32]).float()
+        return input_text, input_video
+
+    def test_forward(self, utils):
+        input_text, input_video = utils
+        clip = videoclip()
+        clip_output = clip(features_a=input_text, features_b=input_video)
+        assert_expected(
+            actual=clip_output.embeddings_a.shape, expected=torch.Size([2, 256])
+        )
+        assert_expected(
+            actual=clip_output.embeddings_b.shape, expected=torch.Size([2, 256])
+        )
+
+    def test_text_dim(self, utils):
```

Comment:
why not adding `test_video_dim`?

Reply:
I originally added `test_text_dim` because the text encoder output dim is configurable and `build_videoclip` relied on that logic. Now that the output dim logic is contained within the text encoder, I don't think this test case is needed anymore (dimensions are checked in `test_forward` already)

Reply:
* to clarify I don't think `test_text_dim` is needed anymore and similarly no `test_video_dim`